atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCache.java

```diff
@@ -36,13 +36,13 @@
     @Override
     void processStartTransactions(Set<Long> startTimestamps);
 
-    @Override
-    void updateCacheAndRemoveTransactionState(Set<Long> startTimestamps);
-
     /**
-     * This method should only be used when the transaction has failed and the state needs to be removed from the
-     * cache without updating the central value store.
+     * This does *not* remove state from the cache - {@link #removeTransactionState(long)} must be called at the end
```

Comment:
Almost makes it sensible to make `TransactionScopedCache` closeable, with a hook to remove itself from the cache when the transaction commits/aborts. It's ugly, but does not rely on future users remembering to use this correctly

Reply:
I'll leave that as a future refactor, for now, since I think that will complicate the next PRs. Maybe in the cleanup pass I'll do this.